apps/parsley/src/hooks/useSections/utils.ts

```diff
@@ -41,68 +56,97 @@ interface SectionEntry {
  * @returns The updated accumulated section data after processing the current line.
  */
 const reduceFn = (
-  accum: SectionEntry[],
+  accum: SectionData,
   line: string,
   logIndex: number,
-): SectionEntry[] => {
+): SectionData => {
   const currentLine = processLine(line);
   // Skip if the current line does not indicate a section
   if (!currentLine) {
     return accum;
   }
-  const sections = [...accum];
+  const { commands, functions } = {
+    commands: [...accum.commands],
+    functions: [...accum.functions],
```

Comment:
Is it necessary to create a new array?
```suggestion
    commands: accum.commands,
    functions: accum.functions,
```